collated/test/A0135782Y.md

```diff
@@ -0,0 +1,533 @@
+    }    
+    
+    @Test
+    public void execute_add_recurringTask_FromDateToDate_unsuccessful_add_as_nonfloating_instead() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        TaskMaster expectedTM = new TaskMaster();
+        Task toAdd = helper.nonFloatingFromDateToDate();
+        expectedTM.addTask(toAdd);
+        RecurringTaskManager.getInstance().correctAddingOverdueTasks(toAdd);
+        List<TaskComponent> expectedComponentList = helper.buildTaskComponentsFromTaskList(expectedTM.getTasks());
+        assertCommandBehavior("add non floating task from XXXX to XXXX from 19 oct 10pm to 20 oct 11am dai t/tag1 t/tag2", 
+                String.format(AddNonFloatingCommand.MESSAGE_SUCCESS, toAdd),
+                expectedTM,
+                expectedComponentList);
+    }
+    
+    @Test
+    public void execute_add_recurringTask_FromDateToDate_successful() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        TaskMaster expectedTM = new TaskMaster();
+        Task toAdd = helper.nonFloatingRecurringFromDateToDate(RecurringType.DAILY);
+        expectedTM.addTask(toAdd);
+        RecurringTaskManager.getInstance().correctAddingOverdueTasks(toAdd);
+        List<TaskComponent> expectedComponentList = helper.buildTaskComponentsFromTaskList(expectedTM.getTasks());
+        assertCommandBehavior("add non floating task from XXXX to XXXX from 19 oct 10pm to 20 oct 11am daily t/tag1 t/tag2",
+                String.format(AddNonFloatingCommand.MESSAGE_SUCCESS, toAdd),
+                expectedTM,
+                expectedComponentList);
+    }
+    
+    @Test
+    public void execute_add_recurringTask_daily_FromDateToDate_daily_caseInsensitive() throws Exception {
+        TestDataHelper helper = new TestDataHelper();
+        TaskMaster expectedTM = new TaskMaster();
+        Task toAdd = helper.nonFloatingRecurringFromDateToDate(RecurringType.DAILY);
+        expectedTM.addTask(toAdd);
+        RecurringTaskManager.getInstance().correctAddingOverdueTasks(toAdd);
+        List<TaskComponent> expectedComponentList = helper.buildTaskComponentsFromTaskList(expectedTM.getTasks());
+        assertCommandBehavior("add non floating task from XXXX to XXXX from 19 oct 10pm to 20 oct 11am dAIly t/tag1 t/tag2",
+                String.format(AddNonFloatingCommand.MESSAGE_SUCCESS, toAdd),
+                expectedTM,
+                expectedComponentList);
+    }
+```
+###### \java\seedu\address\model\RecurringTaskManagerTest.java
+``` java
+public class RecurringTaskManagerTest {
+    private RecurringTaskManager recurringManager;
+    private TaskMaster taskMaster;
+    
+    @Before
+    public void setup() {
+        recurringManager = RecurringTaskManager.getInstance();
+        taskMaster = new TaskMaster();
+        recurringManager.setTaskList(taskMaster.getUniqueTaskList());
+    }
+    
+    @Test
+    public void set_null_tasklist_throwAssert() {
```

Comment:
Refer the test method naming convention again. The name should have no more than three parts.